src/wasm/wasm-validator.cpp

```diff
@@ -209,21 +209,9 @@ struct FunctionValidator : public WalkerPass<PostWalker<FunctionValidator>> {
 
   FunctionValidator(ValidationInfo* info) : info(*info) {}
 
-  struct BreakInfo {
-    enum { UnsetArity = Index(-1), PoisonArity = Index(-2) };
-
-    Type type;
-    Index arity;
-    BreakInfo() : arity(UnsetArity) {}
-    BreakInfo(Type type, Index arity) : type(type), arity(arity) {}
-
-    bool hasBeenSet() {
-      // Compare to the impossible value.
-      return arity != UnsetArity;
-    }
-  };
-
-  std::unordered_map<Name, BreakInfo> breakInfos;
+  // TODO: Switch to std::unordered_set once types are properly canonicalized
+  // so determinism isn't an issue.
```

Comment:
I don't think determinism matters in the validator? Reordered validation errors don't seem that bad.

Reply:
Sure, I guess it would matter more if we had more tests for validation failures. I'll just update this now.